openedx/core/djangoapps/course_live/views.py

```diff
@@ -207,3 +211,69 @@ def get_provider_data(course_id: str) -> Dict:
                 "available": AVAILABLE_PROVIDERS
             }
         }
+
+
+class CourseLiveIFrameView(APIView):
+    """
+    A view for retrieving course live iFrame.
+
+    Path: ``api/course_live/iframe/{course_id}/``
+
+    Accepts: [GET]
+
+    ------------------------------------------------------------------------------------
+    GET
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a program live zoom iframe.
+        * 401: The requesting user is not authenticated.
+        * 404: The requesting user lacks access to the course.
+        * 404: The requested program does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will be iframe HTML.
+
+    **Example**
+
+        {
+            "iframe": "
+                        <iframe
+                            id='lti-tab-embed'
+                            style='width: 100%; min-height: 800px; border: none'
+                            srcdoc='{srcdoc}'
+                            >
+                        </iframe>
+                        ",
+        }
+
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    @ensure_valid_course_key
+    @verify_course_exists()
+    def get(self, request, course_id: str, **_kwargs) -> Response:
+        """
+        Handle HTTP/GET requests
+        """
+        course_key = CourseKey.from_string(course_id)
+        is_lti_enabled = CourseLiveConfiguration.is_enabled(course_key)
+        if not is_lti_enabled:
+            error_data = {
+                "developer_message": "LTI configuration is not available for this course."
+            }
+            return Response(error_data, status=status.HTTP_200_OK)
+
+        course = get_course_with_access(request.user, 'load', course_key)
+        iframe = LtiCourseLaunchMixin().render_to_fragment(request, course)
```

Comment:
I think it makes more sense to inherit from `LtiCourseLaunchMixin`. I think you will also need to override the `_get_lti_roles` method to not include Instructor role. That is a discussion specific role.